### PR TITLE
Week-3 - Server Programming (Steps 4 & 5)

### PR DESCRIPTION
**Pull request by ralucatudor**
Add "Add new comment" feature and use Datastore to store the comments.

Step 4 & 5: To complete the comments feature, I added a `<form>` element to the "Comments" section for **adding new comments** & implemented **doPost()** function to DataServlet.java which takes the data from the request and saves it in the database -> from the doPost() function, after recording the data, I redirected to `#comments` as discussed. I also added location hashes to each section, so in the live server (linked below) you can see that the url changes (corresponding to the section).
I used **Datastore** to permanently store the comments.

[Live server link](http://rtudor-step-2020.appspot.com)

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -48,4 +61,21 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
+
+    Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("text", text);
+    commentEntity.setProperty("author", author);
+    commentEntity.setProperty("date", date);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(commentEntity);
+
+    response.sendRedirect("/#comments");
```
Comment (ChangchengXiao) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:86]: Please add a comment to explain why you would like to do this.

Comment (ralucatudor) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:86]: Done! I added a few comments to both doGet() and doPost()

**Review by ChangchengXiao — APPROVED** [review]

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -34,53 +54,55 @@ sections.forEach((sectionName) => {
...
-    imageContainer.setAttribute("class", "image-container")
+    const imageContainer = document.createElement('div');
+    imageContainer.setAttribute('class', 'image-container')
 
     const totalImagesNumber = 6;
     
     for (let imageIndex = 1; imageIndex <= totalImagesNumber; imageIndex++) {
-        const imgUrl = "images/image-" + imageIndex + ".jpg";
-        const imgElement = document.createElement("img");
+        const imgUrl = 'images/image-' + imageIndex + '.jpg';
```
Comment (skopecki) [inline portfolio/src/main/webapp/script.js]: I'm not feeling strongly about this, but I wanted to point out that you could use `images/image-${ imageIndex}.jpg` to compose strings as well.

**Review by skopecki — APPROVED** [review]

File: portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (ralucatudor) [reply portfolio/src/main/webapp/script.js]: Indeed! I changed my code because I think it looks cleaner with `...${...}...`

**Pull request merged**